src/lib/context/FormContext.tsx

```diff
@@ -0,0 +1,37 @@
+import { createContext, useContext } from "react";
+import { FieldValues, UseFormReturn } from "react-hook-form";
+
+export interface FormContextProps<T extends FieldValues> extends UseFormReturn<T, unknown> {
+  requiredFields: (keyof T)[];
+  disabled: boolean;
+}
+
+export interface InternalFormContextProviderProps<T extends FieldValues> extends Pick<FormContextProps<T>, "requiredFields" | "disabled"> {
+  children: React.ReactNode;
+  formMethods: UseFormReturn<T, unknown>;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const FormContext = createContext<FormContextProps<any>>({} as FormContextProps<any>);
+
+export const InternalFormProvider = <T extends FieldValues>(props: InternalFormContextProviderProps<T>) => {
```

Comment:
This is unused and can be removed.

Reply:
Than you can also remove `InternalFormContextProviderProps`

Reply:
@neotrow I'm a stupid,  fixed :D